src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,691 @@
+# Copyright 2025 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+"""TdnfPackageManager for Azure Linux"""
+import json
+import os
+import re
+
+from core.src.core_logic.VersionComparator import VersionComparator
+from core.src.package_managers.PackageManager import PackageManager
+from core.src.bootstrap.Constants import Constants
+
+
+class TdnfPackageManager(PackageManager):
+    """Implementation of Azure Linux package management operations"""
+
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer, status_handler):
+        super(TdnfPackageManager, self).__init__(env_layer, execution_config, composite_logger, telemetry_writer, status_handler)
+        # Repo refresh
+        self.cmd_clean_cache = "sudo tdnf clean expire-cache"
+        self.cmd_repo_refresh_template = "sudo tdnf -q list updates"
```

Comment:
Why is this called a template? It's only ever used as-is.

Reply:
Addressed